src/Raven.Embedded/RavenEmbedded.cs

```diff
@@ -0,0 +1,212 @@
+
+            return (await server.Value.ConfigureAwait(false)).ServerUrl;
+        }
+
+        private void KillSlavedServerProcess(Process process)
+        {
+            if (process == null || process.HasExited != false)
+                return;
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"Killing global server PID { process.Id }.");
+
+            try
+            {
+                process.Kill();
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                {
+                    Logger.Info($"Failed to kill process {process.Id}", e);
+                }
+            }
+        }
+
+        private async Task<(Uri ServerUrl, Process ServerProcess)> RunServer(ServerOptions options)
+        {
+            var process =  RavenServerRunner.Run(options);
+            if (Logger.IsInfoEnabled)
+            {
+                Logger.Info($"Starting global server: { process.Id }");
+            }
+
+            string url = null;
+            var output = process.StandardOutput;
+            var sb = new StringBuilder();//TODO: listen to standard error as well
+
+            var startupDuration = Stopwatch.StartNew();
+
+            Task<string> readLineTask = null;
+            while (true)
+            {
+                if (readLineTask == null)
+                    readLineTask = output.ReadLineAsync();
+
+                var hasResult = await readLineTask.WaitWithTimeout(TimeSpan.FromSeconds(5)).ConfigureAwait(false);
+
+                if (startupDuration.Elapsed > options.MaxServerStartupTimeDuration) 
+                    break;
+
+                if (hasResult == false)
+                    continue;
+
+                var line = readLineTask.Result;
+
+                readLineTask = null;
+
+                sb.AppendLine(line);
+
+                if (line == null)
+                {
+                    KillSlavedServerProcess(process);
+
+                    throw new InvalidOperationException("Unable to start server, log is: " + Environment.NewLine + sb);
+                }
+                const string prefix = "Server available on: ";
+                if (line.StartsWith(prefix))
+                {
+                    url = line.Substring(prefix.Length);
+                    break;
+                }
+            }
+
+            if (url == null)
+            {
+                var log = sb.ToString();
+
+                KillSlavedServerProcess(process);
+
+                throw new InvalidOperationException("Unable to start server, log is: " + Environment.NewLine + log);
+            }
+
+            return (new Uri(url), process);
+        }
+
+        public void Dispose()
+        {
+            var lazy = Interlocked.Exchange(ref _serverTask, null);
+            if (lazy == null || lazy.IsValueCreated == false)
+                return;
+
+            var process = lazy.Value.Result.ServerProcess;
```

Comment:
Should we use ExceptionAggregator here?

Reply:
That is server side concept, and I think not. Most of the time there is going to be just one, and we don't really expect many errors during disposal